Storage/SQL/BrowserDB.swift

```diff
@@ -51,18 +54,22 @@ open class BrowserDB {
         case Full = 13
     }
 
-    public init(filename: String, secretKey: String? = nil, files: FileAccessor) {
-        log.debug("Initializing BrowserDB: \(filename).")
+    public init(filename: String, secretKey: String? = nil, schema: Schema, files: FileAccessor) throws {
+        self.schema = schema
         self.files = files
         self.filename = filename
         self.secretKey = secretKey
 
+        log.debug("Initializing BrowserDB: \(filename).")
+
         let file = URL(fileURLWithPath: (try! files.getAndEnsureDirectory())).appendingPathComponent(filename).path
-        self.db = SwiftData(filename: file, key: secretKey, prevKey: nil)
 
         if AppConstants.BuildChannel == .developer && secretKey != nil {
             log.debug("Will attempt to use encrypted DB: \(file) with secret = \(secretKey ?? "nil")")
         }
+
+        self.db = SwiftData(filename: file, key: secretKey, prevKey: nil)
+        try reopenIfClosed()
```

Comment:
You're using `reopenIfClosed` for the initial opening! That's… kinda gross. Can we do better, even if only naming-wise?